{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "574fb33a-f3b5-477b-8c00-e139267063dc",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "# BNFO 262 / BIOM 262 / BGGN237 Winter 2026\n",
    "\n",
    "# Homework 1: Statistics and ChIP-seq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8f7f4f6-fbde-41f1-8afd-03ee0ef64232",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "**Name**:\n",
    "\n",
    "**PID**:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91dba4be-5ca1-49ef-8de6-875f23ede145",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "## Instructions\n",
    "\n",
    "Answer the following questions in your own words using a Jupyter notebook and upload a PDF of your answers to Gradescope. Make sure to write your name and PID at the start of your answers. This assignment is due on **Thursday, 01/29/2026 at 9:00AM**."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d64a795a-97e0-46fc-a0ba-99e5a4868b4d",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "### Part 0: (2 Points) Please make sure you select the correct pages for your answer to each question on Gradescope. You will get full points if you assign the pages correctly for all questions."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f605f00c-05f8-4c60-8fb4-2d1afd5e48af",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "## Part 1: Statistics (12 Points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78465e2c-95b1-4041-af56-641e0f4cd352",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "For this section, launch the stats DataHub environment \n",
    "```ghcr.io/biom262/cmm262-notebook:stats```\n",
    "\n",
    "The following questions refer to the following file: \n",
    "```~/public/hw1/colossus_pitch.csv```\n",
    "\n",
    "Note: you don’t need to copy over the file; you can just use the path above.\n",
    "\n",
    "For questions that require plotting and calculations, please include both your final answers and the code you used to obtain them. Submissions with answers only may not receive full credit.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c447d03d-ce3b-4cd1-be12-d56995e32993",
   "metadata": {
    "id": "c447d03d-ce3b-4cd1-be12-d56995e32993"
   },
   "source": [
    "## Questions\n",
    "\n",
    "*Flora colossus* is a plant-like organism native to the planet X. This species is capable of human speech but their vocabulary consists solely of three words: \"I\", \"am\", and their name. It is thought that *F. colossus* can communicate a vast array of complex thoughts and expressions through the pitch and intonation of these three words, but an exact translation has not yet been thoroughly explored.\n",
    "\n",
    "In an attempt at decoding these rules, scientists asked an F. colossus named Groot to translate a set of 50 words carrying either a negative (neg) or positive (pos) connotation (conn). They recorded the responses to each word, and extracted the highest pitch from each response. Their data appears in the file ```colossus_pitch.csv```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81a18af4-2187-43a5-ba14-659ca342d546",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import the necessary packages\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import scipy.stats as stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "931d2d87-e45b-4b38-ad6a-6eb12d2089a5",
   "metadata": {
    "id": "931d2d87-e45b-4b38-ad6a-6eb12d2089a5",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# let's load the data into a dataframe named \"df\"\n",
    "df = pd.read_csv(\"~/public/hw1/colossus_pitch.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d711116-480b-45e4-8470-cee37dbcbd0c",
   "metadata": {
    "id": "3d711116-480b-45e4-8470-cee37dbcbd0c"
   },
   "source": [
    "### Q1 (3 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abe61ce9-04e3-47a2-b7e9-1738ecbf6d4c",
   "metadata": {
    "id": "abe61ce9-04e3-47a2-b7e9-1738ecbf6d4c"
   },
   "source": [
    "Plot two histograms for the pitches in response to 1) the negative words and 2) the positive words. Please label your axes. Based on the plots, do you think there is a significant difference between the two variables? Justify your reasoning."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a7bcb1d-4296-4795-b4d8-439d67fc168f",
   "metadata": {
    "id": "8a7bcb1d-4296-4795-b4d8-439d67fc168f"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5291ef68-8591-4e46-b17e-17b4e4e13fa2",
   "metadata": {
    "id": "5291ef68-8591-4e46-b17e-17b4e4e13fa2",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "49e22ab9-0217-4a21-83c8-a3a3022a0c5b",
   "metadata": {
    "id": "49e22ab9-0217-4a21-83c8-a3a3022a0c5b"
   },
   "source": [
    "### Q2 (2 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a060d8e7-f5b9-4c59-a114-6b82676da6a3",
   "metadata": {
    "id": "a060d8e7-f5b9-4c59-a114-6b82676da6a3"
   },
   "source": [
    "Report the mean and standard deviation of the pitches for 1) the negative words and 2) the positive words."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78301589-5d6c-4087-b798-7f2872635745",
   "metadata": {
    "id": "78301589-5d6c-4087-b798-7f2872635745",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "a093b4c0-384a-4d9d-87cb-0c3ff539e879",
   "metadata": {
    "id": "a093b4c0-384a-4d9d-87cb-0c3ff539e879"
   },
   "source": [
    "### Q3 (2 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b5217f2-17a7-4d51-8c3c-0b2b01824c14",
   "metadata": {
    "id": "9b5217f2-17a7-4d51-8c3c-0b2b01824c14"
   },
   "source": [
    "The scientists think that *F. colossus* will respond with a lower pitch to words carrying a negative connotation and a higher pitch to words carrying a positive connotation. Use your results from the previous question to calculate a z-score and the p-value (using the ```stats.norm.cdf``` function) for this hypothesis. Under the threshold α=0.05, is this a significant result? Was the scientists’ hypothesis correct?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6d51f93-abd8-4b9d-92c1-420211cea703",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4baf6c56-a558-4c83-beb6-31cfc6164e43",
   "metadata": {
    "id": "4baf6c56-a558-4c83-beb6-31cfc6164e43",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b2a0b503-ea68-46c7-bbca-8e526b84d569",
   "metadata": {
    "id": "b2a0b503-ea68-46c7-bbca-8e526b84d569"
   },
   "source": [
    "### Q4 (2 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0dd37635-93d8-48f0-bc69-b79f0baeed6f",
   "metadata": {
    "id": "0dd37635-93d8-48f0-bc69-b79f0baeed6f"
   },
   "source": [
    "What would be the scientists’ hypothesis that makes a two-sided z-test appropriate? State the hypothesis, calculate a z-score and the p-value (using the stats.norm.cdf function) for this hypothesis. Under the threshold α=0.05, is this a significant result? Was the scientists’ hypothesis correct?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4329fd24-2642-4b3b-a7ab-626ef5a58f3a",
   "metadata": {
    "id": "4329fd24-2642-4b3b-a7ab-626ef5a58f3a"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6fcae47-66b4-40d1-8476-b0884d460138",
   "metadata": {
    "id": "c6fcae47-66b4-40d1-8476-b0884d460138",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6015dab7-48d7-4061-9065-e46da5e161d8",
   "metadata": {
    "id": "b2a0b503-ea68-46c7-bbca-8e526b84d569"
   },
   "source": [
    "### Q5 (3 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d6e00f8-61c3-44f2-8365-bb02b2333d4b",
   "metadata": {
    "id": "0dd37635-93d8-48f0-bc69-b79f0baeed6f"
   },
   "source": [
    "Let's say that we decided to use a t-test instead of a z-test. What kind of distribution are we assuming that the sample means (of the pitches) follow now? Are we now more or less likely to reject the null hypothesis? For full credit, please justify your response. Hint: sketch out the distributions on paper."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1be47243-b847-4df5-9f5a-32bc8764be42",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "528f4adf-a3f4-4fae-b3a6-cfbc08f8f94b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5f3f8126-129f-44ba-b787-f40afe867b6b",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "## Part 2: ChIP-seq (7 Points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f606e82-e21c-4d96-ade8-78a62c38fc64",
   "metadata": {
    "id": "574fb33a-f3b5-477b-8c00-e139267063dc"
   },
   "source": [
    "For this section, launch the ChIP-seq DataHub environment \n",
    "```ghcr.io/biom262/cmm262-notebook:chipseq```\n",
    "\n",
    "Open a terminal and run ```source activate chipseq```\n",
    "\n",
    "The following questions refer to the following file: \n",
    "```\n",
    "~/public/hw1/hw-f1.bam \n",
    "~/public/hw1/hw-f2.bam \n",
    "```\n",
    "Note: you don’t need to copy over the file; you can just use the path above.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95cdbaae-ee78-41b1-a206-2899f0868429",
   "metadata": {
    "id": "3d711116-480b-45e4-8470-cee37dbcbd0c"
   },
   "source": [
    "### Q1 (2 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b854592-e50d-417c-b41c-0a4b149fd2c5",
   "metadata": {},
   "source": [
    "You received two BAM files from a colleague (hw-f1.bam and hw-f2.bam) but there was a mess up, and the files were not annotated properly. Which UNIX command(s) might you use to investigate which species the data was aligned to? Justify why your command(s) would work."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25ae4a35-36fd-4068-98bd-855ddb44b038",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc9ec620-2c9c-4cab-9ba0-df000ab13cb7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5b3a53c7-03ae-46e0-a0d3-51452b3e8e24",
   "metadata": {
    "id": "3d711116-480b-45e4-8470-cee37dbcbd0c"
   },
   "source": [
    "### Q2 (1 point)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d1b0e28-fb80-403d-b657-636ae305db2f",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "Find the two BAM files in the hw1 directory and then run your suggested commands from the previous question. Which species were the two BAM files aligned to?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51a77aeb-bd7a-4e81-8c29-dc2fa4798c13",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d798f964-7ace-4d46-a335-62bc25f9d972",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "d58d3495-44af-47f8-babe-dedb4f3c5f10",
   "metadata": {
    "id": "3d711116-480b-45e4-8470-cee37dbcbd0c"
   },
   "source": [
    "### Q3 (1 point)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f3d15f3-adab-46c1-850e-94f56758f356",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "What chromosome does the data in these two files come from? Note, some reads can align to multiple chromosomes, so make sure that your answer is not impacted by outlier reads."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6605f51-3665-4f39-b723-900853556781",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ce753d9-c28a-4cf4-9824-6a4f18b312b8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f4f9f5b7-2489-4b07-b7af-86775cac9345",
   "metadata": {
    "id": "3d711116-480b-45e4-8470-cee37dbcbd0c"
   },
   "source": [
    "### Q4 (3 points)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c888e6d-6870-47a6-8d7f-621bffa1fc46",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "You were told that the two files are from a ChIP-seq experiment, and you are trying to figure out what is in each file: input or ChIP of DNA-binding protein. Which file is the input and which file comes from the ChIP-seq of the DNA-binding protein? What is the target protein/histone modification? Explain the steps you took and the commands that you used to reach this decision and add screenshots of any visualization tools. Hint: use two consecutive steps of the eleven in the ChIP-seq module."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2d6a430-28b1-490f-9c45-e3a2f7a0d666",
   "metadata": {
    "id": "f6d51f93-abd8-4b9d-92c1-420211cea703"
   },
   "source": [
    "ANSWER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b19bec0-83d9-443f-8386-30c101cd359a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
